effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -829,11 +830,98 @@ object Typer extends Phase[NameResolved, Typechecked] {
           Result((), effBinding)
         }
 
-      case d @ source.ExternDef(pure, id, tps, vps, bps, tpe, body) => Context.withUnificationScope {
+      case d @ source.ExternDef(pure, id, tps, vps, bps, tpe, bodies) => Context.withUnificationScope {
         d.symbol.vparams foreach Context.bind
         d.symbol.bparams foreach Context.bind
 
-        body.args.foreach { arg => checkExpr(arg, None) }
+        bodies.foreach{
+          case source.ExternBody.StringExternBody(ff, body) =>
+            body.args.foreach { arg => checkExpr(arg, None) }
+          case source.ExternBody.EffektExternBody(ff, body) =>
+            // Check with each body as if this was a function definition
+            val sym = d.symbol
+            // was assigned by precheck
+            val functionCapture = Context.lookupCapture(sym)
+
+            // We can also try to solve for the function capture, after checking the function.
+            // Hence we provide it to `withUnificationScope`.
+            val captVars = functionCapture match {
+              case x: CaptUnificationVar => List(x)
+              case _ => Nil
+            }
+            val Result(funTpe, unhandledEffects) = Context.withUnificationScope(captVars) {
+
+              sym.vparams foreach Context.bind
+              sym.bparams foreach Context.bind
+
+              // to subtract the capabilities, which are only inferred bottom up, we need a **second** unification variable
+              val inferredCapture = Context.freshCaptVar(CaptUnificationVar.AnonymousFunctionRegion(BlockLiteral(tps, vps, bps, body)))
```

Comment:
Apart from the role here, all of this code is just copy-pasted from the case for `FunDef`.

Reply:
That's a lot of code. Can we somehow share?

Reply:
We probably could. Thinking about it, frankly, I'm not sure this is the correct way to check those.
We probably want at least "internal" type checking, but I'm not quite certain how much we want at the boundary.

Will look at it again once I find the time.